src/trace-span.js

```diff
@@ -37,7 +37,13 @@ function TraceSpan(name, spanId, parentSpanId) {
  * @param {string} value The new value of the label.
  */
 TraceSpan.prototype.setLabel = function(key, value) {
-  this.labels[key] = '' + value;
+  var val = '' + value;
+  // The official limit imposed by the service is 16k per label value.
+  // We truncate smaller than that to keep requests small.
+  if (val.length > 5000) {
```

Comment:
It would be better to make this an (internal) config property rather than a hard-coded constant.